tests/cross_functional/system_test/test_graceful_nodes_shutdown.py

```diff
@@ -0,0 +1,439 @@
+        snap_obj = snapshot_factory(pvc_obj)
+        return pvc_obj, pod_obj, file_name, orig_md5_sum, snap_obj
+
+    def setup_encrypted_pvc(
+        self,
+        pvc_type="block",
+        pv_encryption_kms_setup_factory=None,
+        storageclass_factory=None,
+        interface=constants.CEPHFILESYSTEM,
+        pvc_factory=None,
+        pod_factory=None,
+        pvc_clone_factory=None,
+        snapshot_factory=None,
+    ):
+        """
+        encrypted block/fs pvc
+        Returns:
+            pvc object
+            pod object
+            file_name
+            origin md5sum
+            snapshot
+        """
+
+        logger.info(f"Adding encrypted {pvc_type} pvc")
+        if interface == constants.CEPHBLOCKPOOL:
+            self.vault = pv_encryption_kms_setup_factory("v1", False)
+            self.sc_obj = storageclass_factory(
+                interface=interface,
+                encrypted=True,
+                encryption_kms_id=self.vault.kmsid,
+            )
+            self.vault.vault_path_token = self.vault.generate_vault_token()
+            self.vault.create_vault_csi_kms_token(namespace=self.proj_obj.namespace)
+        pvc_obj = pvc_factory(
+            interface=interface,
+            storageclass=self.sc_obj,
+            size=3,
+            status=constants.STATUS_BOUND,
+            project=self.proj_obj,
+        )
+        pod_obj = pod_factory(
+            interface=interface,
+            pvc=pvc_obj,
+            status=constants.STATUS_RUNNING,
+        )
+        # Run IO for data integrity
+        file_name = pod_obj.name
+        logger.info(f"File created during IO {file_name} (encrypted {pvc_type} pvc)")
+        pod_obj.run_io(storage_type="fs", size="500M", fio_filename=file_name)
+
+        # Wait for fio to finish
+        pod_obj.get_fio_results()
+        logger.info(f"Io completed on pod {pod_obj.name} (encrypted {pvc_type} pvc)")
+
+        origin_md5sum = pod.cal_md5sum(pod_obj, file_name)
+
+        pvc_clone_factory(pvc_obj)
+        encrypt_snap_obj = snapshot_factory(pvc_obj)
+
+        return pvc_obj, pod_obj, file_name, origin_md5sum, encrypt_snap_obj
+
+    def validate_snapshot_restore(self, snapshot_restore_factory):
+        """
+        Verifies the snapshot restore works fine
+        Raises: If failed restore snapshot
+        """
+        logger.info("Creating snapshot restore for non-encrypted pvcs after reboot")
+        ne_restored_pvc = snapshot_restore_factory(
+            snapshot_obj=self.snap_obj,
+            volume_mode=self.snap_obj.parent_volume_mode,
+        )
+        eb_restored_pvc = snapshot_restore_factory(
+            snapshot_obj=self.eb_snap_obj,
+            volume_mode=self.snap_obj.parent_volume_mode,
+        )
+        efs_restored_pvc = snapshot_restore_factory(
+            snapshot_obj=self.efs_snap_obj,
+            volume_mode=self.snap_obj.parent_volume_mode,
+        )
+
+        self.validate_pvc_expansion(
```

Comment:
the function name says validate_snapshot_restore but we are also doing pvc expansion validation? can you move this since it is not a relevant line of code according to the func name

Reply:
In https://docs.google.com/document/d/129IreytPAPPQW1sov1JyQ6_SHXfgpVF4EtD9ZP9m10s/edit we added pvc expansion check after restore snapshot in post condition check.